tarantool/src/main/java/com/yahoo/ycsb/db/TarantoolClient.java

```diff
@@ -0,0 +1,168 @@
+			this.connection = new TarantoolConnection16Impl(host, port);
+		}
+		catch (Exception exc) {
+			System.err.println("Can't init Tarantool connection:" + exc.toString());
+			exc.printStackTrace();
+			return;
+		}
+	}
+	
+	public void cleanup() throws DBException{
+		this.connection.close();
+	}
+
+	@Override
+	public int insert(String table, String key, HashMap<String, ByteIterator> values) {
+		int j = 0;
+		String[] tuple = new String[1 + 2 * values.size()];
+		tuple[0] = key;
+		for (Map.Entry<String, ByteIterator> i: values.entrySet()) {
+			tuple[j + 1] = i.getKey();
+			tuple[j + 2] = i.getValue().toString();
+			j += 2;
+		}
+		try {
+			this.connection.insert(this.spaceNo, tuple);
+		} catch (TarantoolException e) {
+			e.printStackTrace();
+			return 1;
+		}
+		return 0;
+	}
+
+	private HashMap<String, ByteIterator> tuple_convert_filter (List<String> input,
+			Set<String> fields) {
+		HashMap<String, ByteIterator> result = new HashMap<String, ByteIterator>();
+		if (input == null)
+			return result;
+		for (int i = 1; i < input.toArray().length; i += 2)
+			if (fields == null || fields.contains(input.get(i)))
+				result.put(input.get(i), new StringByteIterator(input.get(i+1)));
+		return result;
+	}
+
+	@Override
+	public int read(String table, String key, Set<String> fields,
+			HashMap<String, ByteIterator> result) {
+		try {
+			List<String> response;
+			response = this.connection.select(this.spaceNo, 0, Arrays.asList(key), 0, 1, 0);
+			result = tuple_convert_filter(response, fields);
+			return 0;
+		} catch (TarantoolException e) {
+			e.printStackTrace();
+			return 1;
+		} catch (IndexOutOfBoundsException e) {
+			return 1;
+		}
+	}
+
+	@Override
+	public int scan(String table, String startkey,
+			int recordcount, Set<String> fields,
+			Vector<HashMap<String, ByteIterator>> result) {
+		List<String> response;
+		try {
+			response = this.connection.select(this.spaceNo, 0, Arrays.asList(startkey), 0, recordcount, 6);
+		} catch (TarantoolException e) {
+			e.printStackTrace();
+			return 1;
+		}
+		HashMap<String, ByteIterator> temp = tuple_convert_filter(response, fields);
+		if (!temp.isEmpty())
+			result.add((HashMap<String, ByteIterator>) temp.clone());
+		return 0;
+	}
+
+	@Override
+	public int delete(String table, String key) {
+		try {
+			this.connection.delete(this.spaceNo, Arrays.asList(key));
+		} catch (TarantoolException e) {
+			e.printStackTrace();
+			return 1;
+		} catch (IndexOutOfBoundsException e) {
+			return 1;
+		}
+		return 0;
+	}
+	@Override
+	public int update(String table, String key,
+			HashMap<String, ByteIterator> values) {
+		int j = 0;
+		String[] tuple = new String[1 + 2 * values.size()];
+		tuple[0] = key;
+		for (Map.Entry<String, ByteIterator> i: values.entrySet()) {
+			tuple[j + 1] = i.getKey();
+			tuple[j + 2] = i.getValue().toString();
+			j += 2;
+		}
+		try {
+			this.connection.replace(this.spaceNo, tuple);
+		} catch (TarantoolException e) {
+			e.printStackTrace();
```

Comment:
nit: as above, log the exception via your logger rather than printing the stacktrace yourself